compiler/src/dotty/tools/dotc/reporting/diagnostic/messages.scala

```diff
@@ -1223,4 +1224,26 @@ object messages {
            |Attempting to define a field in a method signature after a varargs field is an error.
            |""".stripMargin
   }
+
+  case class AmbiguousOverload(tree: tpd.Tree, alts: List[SingleDenotation], pt: Type)(
+    err: typer.ErrorReporting.Errors)(
+    implicit ctx: Context)
+  extends Message(AmbiguousOverloadID) {
+
+    private def all = if (alts.length == 2) "both" else "all"
+
+    override def msg: String =
+      s"""|Ambiguous overload. The ${err.overloadedAltsStr(alts)}
+          |$all match ${err.expectedTypeStr(pt)}""".stripMargin
+
+    override def kind: String = "Reference"
+
+    override def explanation: String =
+       s"""|There are ${alts.length} methods that could be referenced as the compiler knows too little about the expected type.
+           |You may specify the expected type e.g. by
+           |- assigning it to a value with a specified type, or
+           |- adding a type ascription as in `(instance.myMethod: String => Int)`
```

Comment:
`tpe.show` gives `(s: String)Int`, but I'd like to get `(String) => Int` for better cut-and-paste fidelity.